pkg/skaffold/deploy/helm/deploy.go

```diff
@@ -97,7 +108,7 @@ type Config interface {
 }
 
 // NewDeployer returns a configured Deployer.  Returns an error if current version of helm is less than 3.0.0.
-func NewDeployer(cfg Config, labels map[string]string, h *latestV1.HelmDeploy) (*Deployer, error) {
+func NewDeployer(cfg Config, labels map[string]string, logProvider log.Provider, previewProvider preview.Provider, debugProvider debug.Provider, h *latestV1.HelmDeploy) (*Deployer, error) {
```

Comment:
should we have a `DeployerOptions` function parameter instead, like:
```go
type DeployerOptions struct {
    log log.Provider
    preview preview.Provider
    debug debug.Provider
}
```

Reply:
yeah I went back and forth on this, but it looks cleaner this way so let's go for it.